ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -190,28 +190,48 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
             if (scroll) {
                 this.scrollToTop();
             }
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-                this.currentSectionIndex = nextIndex;
-                this.visitedSectionIndexes[nextIndex] = true;
-                this.saveLastVisitedSectionIndex(nextIndex);
-            }
+            this.submitAnnouncementService.announceSubmit(null);
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(() => {
+                            this.validationRequested = true;
+                            this.sendSectionAnalytics();
+                            this.currentSection.validate();
+                            if (this.currentSection.valid) {
+                                this.resetValidationState();
+                                this.visitedSectionIndexes[nextIndex] = true;
+                                this.saveLastVisitedSectionIndex(nextIndex);
+                                this.currentSectionIndex = nextIndex;
+                            }
+                        })
+                    )
+                ),
+                take(1)
+            ).subscribe();
         }
```

Comment:
From activity component this was the big change: what we do when we hit the Next button:
First, we trigger this.submitAnnouncementService.announceSubmit().

The timer(100) line there is to allow the mailing address to (perhaps, becoming "busy" after getting submit announced).

Mailing address component listens for this signal to trigger a "real" saving of the mailing address.

Then we listen for the page not being busy before proceeding to do validations increment steps, etc. Most of that stuff that happens once not busy is pretty much what was there before.

Reply:
Additional changes since last time: had to add additional things to incrementStep to get back the "scroll to error" feature. This was happening before when triggering a form submit, but we are doing less than that. The comments in code describe what is going on.

Reply:
Also added back incrementStep and decrementStep the "isLoaded$" observable updates. Appears that stepper depends on it. Hopefully I did this right?